CHANGELOG.md

```diff
@@ -1,5 +1,14 @@
 # Changelog
 
+## [3.0.4](https://github.com/Pinterest/PINCache/tree/3.0.4) (TBD)
```

Comment:
I can update the version number and release date before merging, per guidance from the maintainers.

Reply:
I'd suggest making this a separate diff. We should look at some other PRs and possibly merge a few more before making a new release.

Reply:
@rcancro Sounds good. 👍 

Would you prefer I revert all my changes to the CHANGELOG?

Reply:
Yes, could you revert all your changes to the CHANGELOG.md. Reading the RELEASE.md runbook, I think that will automatically generate the changes.

Reply:
You bet. 

I'll rebase this PR without changes to the CHANGELOG.md.

Reply:
@andyfinnell Done! I've rebased this PR which no longer includes changes to CHANGELOG.md